### PR TITLE
[unixodbc] no absolute paths

### PR DESCRIPTION
**Pull request by autoantwort**
For #20469

### CODE REVIEW

**Review by LilyWangLL — APPROVED** [review]

File: ports/unixodbc/portfile.cmake [hunk 1]
```diff
@@ -35,6 +35,16 @@ file(REMOVE_RECURSE
      "${CURRENT_PACKAGES_DIR}/debug/etc"
      "${CURRENT_PACKAGES_DIR}/etc"
      "${CURRENT_PACKAGES_DIR}/share/man"
-     )
+)
+
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h" "#define BIN_PREFIX \"${CURRENT_INSTALLED_DIR}/tools/unixodbc/bin\"" "")
```
Comment (BillyONeal) [inline ports/unixodbc/portfile.cmake:40]: These look like they may be 'exported' to customers, can you explain why just deleting them is OK and/or involve upstream?

**Review by BillyONeal — DISMISSED** [review]

File: ports/unixodbc/portfile.cmake (hunk 1 above)
Comment (autoantwort) [reply ports/unixodbc/portfile.cmake:40]: As far as I can tell this value is not used. There is also no documentation about this value.

Comment (LilyWangLL) [reply ports/unixodbc/portfile.cmake:40]: Ping @BillyONeal

Comment (BillyONeal) [reply ports/unixodbc/portfile.cmake:40]: I wish I understood why it emailed me for Lily's reply but not autoantwort's :(.

**Comment by BillyONeal** [comment]
@lurcher This change has the potential to impact you. Are you OK with us patching out these absolute paths or would you like some form of edit to https://github.com/lurcher/unixODBC/ to avoid generating them in the first place? vcpkg ports need to be relocatable in case they're used on a different machine...

**Comment by lurcher** [comment]
On 01/12/2021 02:31, BillyONeal wrote:
>
> @lurcher <https://github.com/lurcher> This change has the potential to 
> impact you. Are you OK with us patching out these absolute paths or 
> would you like some form of edit to 
> https://github.com/lurcher/unixODBC/ 
> <https://github.com/lurcher/unixODBC/> to avoid generating them in the 
> first place? vcpkg ports need to be relocatable in case they're used 
> on a different machine...
>

Have I missed something? What change?

-- 
Nick

**Comment by autoantwort** [comment]
> Have I missed something? What change?

The change of this PR:
```cmake
vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h" "#define BIN_PREFIX \"${CURRENT_INSTALLED_DIR}/tools/unixodbc/bin\"" "")
vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h" "#define DEFLIB_PATH \"${CURRENT_INSTALLED_DIR}/lib\"" "")
vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h" "#define EXEC_PREFIX \"${CURRENT_INSTALLED_DIR}\"" "")
vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h" "#define INCLUDE_PREFIX \"${CURRENT_INSTALLED_DIR}/include\"" "")
vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h" "#define LIB_PREFIX \"${CURRENT_INSTALLED_DIR}/lib\"" "")
vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h" "#define PREFIX \"${CURRENT_INSTALLED_DIR}\"" "")
vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h" "#define SYSTEM_FILE_PATH \"${CURRENT_INSTALLED_DIR}/etc\"" "")
vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h" "#define SYSTEM_LIB_PATH \"${CURRENT_INSTALLED_DIR}/lib\"" "")
```
It removes all defines that are pointing to an absolute path

**Comment by lurcher** [comment]
On 01/12/2021 16:33, autoantwort wrote:
>
>     Have I missed something? What change?
>
> The change of this PR:
>
> vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h"  "#define BIN_PREFIX \"${CURRENT_INSTALLED_DIR}/tools/unixodbc/bin\""  "")
> vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h"  "#define DEFLIB_PATH \"${CURRENT_INSTALLED_DIR}/lib\""  "")
> vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h"  "#define EXEC_PREFIX \"${CURRENT_INSTALLED_DIR}\""  "")
> vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h"  "#define INCLUDE_PREFIX \"${CURRENT_INSTALLED_DIR}/include\""  "")
> vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h"  "#define LIB_PREFIX \"${CURRENT_INSTALLED_DIR}/lib\""  "")
> vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h"  "#define PREFIX \"${CURRENT_INSTALLED_DIR}\""  "")
> vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h"  "#define SYSTEM_FILE_PATH \"${CURRENT_INSTALLED_DIR}/etc\""  "")
> vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h"  "#define SYSTEM_LIB_PATH \"${CURRENT_INSTALLED_DIR}/lib\""  "")
>
> It removes all defines that are pointing to an absolute path
>

Ok, and whats thats going to alter in the unixODBC project? Assume I am 
stupid, as on this subject that's how I feel at the moment :-)

**Comment by autoantwort** [comment]
The question was if this change will break anything (this project or users of this project)

**Comment by lurcher** [comment]
On 01/12/2021 16:52, Nick Gorham wrote:
> On 01/12/2021 16:33, autoantwort wrote:
>>
>>     Have I missed something? What change?
>>
>> The change of this PR:
>>
>> vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h"  "#define BIN_PREFIX \"${CURRENT_INSTALLED_DIR}/tools/unixodbc/bin\""  "")
>> vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h"  "#define DEFLIB_PATH \"${CURRENT_INSTALLED_DIR}/lib\""  "")
>> vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h"  "#define EXEC_PREFIX \"${CURRENT_INSTALLED_DIR}\""  "")
>> vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h"  "#define INCLUDE_PREFIX \"${CURRENT_INSTALLED_DIR}/include\""  "")
>> vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h"  "#define LIB_PREFIX \"${CURRENT_INSTALLED_DIR}/lib\""  "")
>> vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h"  "#define PREFIX \"${CURRENT_INSTALLED_DIR}\""  "")
>> vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h"  "#define SYSTEM_FILE_PATH \"${CURRENT_INSTALLED_DIR}/etc\""  "")
>> vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/unixodbc_conf.h"  "#define SYSTEM_LIB_PATH \"${CURRENT_INSTALLED_DIR}/lib\""  "")
>>
>> It removes all defines that are pointing to an absolute path
>>
>
> Ok, and whats thats going to alter in the unixODBC project? Assume I 
> am stupid, as on this subject that's how I feel at the moment :-)
>
I you mean to make changes to unixodbc_conf.h in the unixODBC 
repository, then you should be aware that the .h file is generated at 
build time from unixodbc_conf.h.in

I guess my question is what the changes are for. The original choices 
were and are based on the GNU configure package layout guidelines.

-- 
Nick

**Comment by autoantwort** [comment]
> I guess my question is what the changes are for.

To make the package relocatable. After the package is build it is put into the binary cache. It than can be restored on another PC on to another location. In this new location all the paths are wrong because they are absolute and are pointing to non existing locations.

**Comment by lurcher** [comment]
On 01/12/2021 19:06, autoantwort wrote:
>
>     I guess my question is what the changes are for.
>
> To make the package relocatable. After the package is build it is put 
> into the binary cache. It than can be restored on another PC on to 
> another location. In this new location all the paths are wrong because 
> they are absolute and are pointing to non existing locations.
>

Ok, and how will that impact on those that want it to point to absolute 
paths?

-- 
Nick

**Comment by lurcher** [comment]
On 01/12/2021 19:06, autoantwort wrote:
>
>     I guess my question is what the changes are for.
>
> To make the package relocatable. After the package is build it is put 
> into the binary cache. It than can be restored on another PC on to 
> another location. In this new location all the paths are wrong because 
> they are absolute and are pointing to non existing locations.
>
>
Thinking for a few seconds more, the only path that unixODBC cares about 
is the location of the ini files at run time. And thats already set (or 
can be) by shell variables. unixodbc_conf.h is there mainly so other 
packages can find where unixODBC is to build against.

-- 
NIck

**Comment by BillyONeal** [comment]
OK, upstream ( @lurcher ) indicates that it's OK to delete these. @lurcher we would recommend you make a change like this for all customers, not only those in vcpkg, but we'll merge @autoantwort 's change for now. (Assuming build comes back OK)

**Pull request merged**